router/authentication/jwks.go

```diff
@@ -0,0 +1,98 @@
+package authentication
```

Comment:
Can we put this package to `internals`?

Reply:
No, otherwise we can't call it from the integration tests

Reply:
Why do we have to? An integration test must not rely on internals.

Reply:
Can we split it out so we don't need to reference the router code?

Reply:
We have no means to set the router configuration from a config file other than `cmd/router/main.go`, but this starts the whole router. For integration tests we call directly into `core.NewRouter()` which requires the arguments to be already parsed. In order to pass an array of `authentication.Authenticator()`, the package needs to be importable by `router-tests`. If we move it to `internal`, then `router-tests` can't pass authenticators.

Reply:
ok